spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerMutationFactoryImpl.java

```diff
@@ -130,13 +132,37 @@ public Mutation delete(Class entityClass, Key key) {
 					.getProperty(spannerPersistentProperty);
 			if (kids != null) {
 				for (Object child : kids) {
+					verifyChildHasParentId(persistentEntity, object,
+							this.spannerMappingContext.getPersistentEntity(
+									spannerPersistentProperty.getColumnInnerType()), child);
 					mutations.addAll(saveObject(op, child, includeColumns));
 				}
 			}
 		});
 		return mutations;
 	}
 
+	private void verifyChildHasParentId(SpannerPersistentEntity parentEntity,
+			Object parentObject, SpannerPersistentEntity childEntity,
+			Object childObject) {
+		SpannerPersistentProperty[] parentPK = parentEntity.getPrimaryKeyProperties();
+		SpannerPersistentProperty[] childPK = childEntity.getPrimaryKeyProperties();
+		PersistentPropertyAccessor parentAccessor = parentEntity
+				.getPropertyAccessor(parentObject);
+		PersistentPropertyAccessor childAccessor = childEntity
+				.getPropertyAccessor(childObject);
+		for (int i = 0; i < parentPK.length; i++) {
+			if (!parentAccessor.getProperty(parentPK[i])
+					.equals(childAccessor.getProperty(childPK[i]))) {
+				throw new SpannerDataException(
+						"A child entity's common primary key columns with its parent must "
+								+ "have the same values. Primary key component " + i
```

Comment:
Can we include the property name in the exception?

Reply:
Yes, added the column name to the message.